### PR TITLE
To Cross verify build version with source version

### PR DESCRIPTION
**Pull request by DivyaJyothiKasam**

### CODE REVIEW

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if [ $? -ne 0 ]; then
+			msg="ERROR: Failed to install wget"
+			LogMsg "$msg"
+			UpdateSummary "$msg"
+			SetTestStateFailed
+			exit 1
+		fi
+    fi
+rm -rf hv_compat.h
+module="hv_vmbus"
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: This variable is not needed. You can directly use the value.

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+version=$(modinfo $module | grep version: | head -1 | awk '{print $2}')
+LogMsg $version
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Better log message:
LogMsg "modinfo detected version: $version"

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+wget https://raw.githubusercontent.com/LIS/lis-next/$version/hv-rhel7.x/hv/include/linux/hv_compat.h
+sourceversion=$(grep 'define HV_DRV_VERSION' hv_compat.h|cut -d '"' -f 2)
+LogMsg $sourceversion
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Better log message:
LogMsg "version found in source code: $sourceversion"

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+rm -rf hv_compat.h
+module="hv_vmbus"
+version=$(modinfo $module | grep version: | head -1 | awk '{print $2}')
+LogMsg $version
+wget https://raw.githubusercontent.com/LIS/lis-next/$version/hv-rhel7.x/hv/include/linux/hv_compat.h
+sourceversion=$(grep 'define HV_DRV_VERSION' hv_compat.h|cut -d '"' -f 2)
+LogMsg $sourceversion	
+if [ "$version" = "$sourceversion" ]
+then
+	echo "Build version and Source version are same"
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: space error

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+			UpdateSummary "$msg"
+			SetTestStateFailed
+			exit 1
+		fi
+    fi
+rm -rf hv_compat.h
+module="hv_vmbus"
+version=$(modinfo $module | grep version: | head -1 | awk '{print $2}')
+LogMsg $version
+wget https://raw.githubusercontent.com/LIS/lis-next/$version/hv-rhel7.x/hv/include/linux/hv_compat.h
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: You need error check for wget also. what if that file is not available?

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+sourceversion=$(grep 'define HV_DRV_VERSION' hv_compat.h|cut -d '"' -f 2)
+LogMsg $sourceversion	
+if [ "$version" = "$sourceversion" ]
+then
+	echo "Build version and Source version are same"
+    UpdateSummary "LIS version check is success"
+    SetTestStateCompleted     
+else
+	LogMsg "LIS Version check failed"
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Lots of tab/space errors

File: XML/TestCases/BVT.xml [hunk 7]
```diff
@@ -1,4 +1,15 @@
 <TestCases>
+    <test>
+        <testName>BVT_VERSION_CHECK</testName>
```
Comment (santoshx) [inline XML/TestCases/BVT.xml]: Underscores in testname?

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Check wget
+wget -V > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+    yum install wget -y
+    if [ $? -ne 0 ]; then
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: You should use check_exit_status function

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 5 above)
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Use double quite around $version

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 3 above)
Comment (prashantmhetar) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: What about #define _HV_DRV_VERSION?

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 5 above)
Comment (prashantmhetar) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: hv_compat.h is available for 5.x, 6.x and 7.x

**Comment by santoshx** [comment]
I think change name should be "Cross verify build version with source version"

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 1 above)
Comment (DivyaJyothiKasam) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: Removed the variable

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 2 above)
Comment (DivyaJyothiKasam) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: Modified the message

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 3 above)
Comment (DivyaJyothiKasam) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: done

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 5 above)
Comment (DivyaJyothiKasam) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: Made it for 3 versions

File: XML/TestCases/BVT.xml (hunk 7 above)
Comment (DivyaJyothiKasam) [reply XML/TestCases/BVT.xml]: Removed underscores

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 8 above)
Comment (DivyaJyothiKasam) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: Done

File: Testscripts/Linux/LIS-VERSION-CHECK.sh (hunk 5 above)
Comment (DivyaJyothiKasam) [reply Testscripts/Linux/LIS-VERSION-CHECK.sh]: Done

**Comment by DivyaJyothiKasam** [comment]
Logs:
[LISAv2 Test Results Summary]
Test Run On           : 01/08/2019 12:54:53
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:6
XML File              : TestConfiguration

   ID TestCaseName                                                 TestResult TestDuration(in minutes)
------------------------------------------------------------------------------------------------------
    1 BVT-VERSION-CHECK                                                  PASS                 6.35

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Use check_exit_status here as well.

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+	LogMsg "$msg"
+	UpdateSummary "$msg"
+	SetTestStateFailed
+	exit 1
+	fi
+fi	
+version=$(modinfo "hv_vmbus" | grep version: | head -1 | awk '{print $2}')
+rm -rf hv_compat.h
+wget https://raw.githubusercontent.com/LIS/lis-next/"$version"/hv-rhel7.x/hv/include/linux/hv_compat.h
+check_exit_status "File hv_compat.h is downloaded" "exit"
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Change to:
check_exit_status  "Download file hv_compact.h" "exit"
Then it'll print like:
Download file hv_compat.h: success
or
Download file hv_compat.h: fail

Do this everywhere

File: XML/TestCases/BVT.xml [hunk 11]
```diff
@@ -446,4 +457,4 @@
         <Tags>wala</Tags>
         <Priority>0</Priority>
     </test>
-</TestCases>
\ No newline at end of file
+</TestCases>
```
Comment (santoshx) [inline XML/TestCases/BVT.xml]: Don't do this change.

Comment (santoshx) [reply XML/TestCases/BVT.xml]: Are we not handling hexadecimal version  _HV_DRV_VERSION  in this test?

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    exit 2
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Check wget
+wget -V > /dev/null 2>&1
+check_exit_status "wget is already installed" "exit"
+if [ $? -ne 0 ];
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Check_exit_status already done this. You don't need if condition again.

File: Testscripts/Linux/LIS-VERSION-CHECK.sh [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+UtilsInit
+
+# Check wget
+wget -V > /dev/null 2>&1
+check_exit_status "wget is already installed" "exit"
+if [ $? -ne 0 ]; 
+then
+	yum install wget -y
+	if [ $? -ne 0 ]; then
+		msg="ERROR: Failed to install wget"
```
Comment (santoshx) [inline Testscripts/Linux/LIS-VERSION-CHECK.sh]: Use check_exit_status

**Comment by DivyaJyothiKasam** [comment]
@prashantmhetar @mboitei   I've resolved all review comments by Santosh & Prashant . Please let me know whether I can proceed to create PR in main repository.?

**Comment by santoshx** [comment]
@DivyaJyothiKasam 
Looks good to me. 
Okay from my side.
Make sure to test it before creating PR in main repository.

**Pull request merged**